### PR TITLE
AMBR-1069

### PR DESCRIPTION
**Pull request by egh**
Testing:

1. Build this using `mvn install` so that it installs the package into `/var/tmp/apt`
1. Build https://github.com/PLOS/rhino/pull/508 using `mvn install`
1. Build https://github.com/PLOS/plos-themes/pull/1020 using `mvn install`
(whew)
1. ssh to `aa-stage.plos.org` and run:
  ```sh
  $ ploscli repackage dev http://journals-stage1-qc1.soma.plos.org:8006/v2   10.1371/journal.pone.0217207 --revision_number=1 --output=pone.0217207.zip
  $ ploscli repackage dev http://journals-stage1-qc1.soma.plos.org:8006/v2 10.1371/journal.pone.0213096 --revision_number=1 --output=pone.0213096.zip
  ```
5. rsync or scp these zip files to your desktop.
1. `cd pogos2/vagrant/apps/journals`
1. Run `vagrant up`
1. Run `mysql -u ambra -h journals.vagrant.test -pambra201 ambra < seed.sql`
1. Run `curl -X POST http://journals.vagrant.test:8002/v1/buckets -d name=corpus`
1. Run `./ingest path/to/pone.0213096.zip`
1. Run `./ingest path/to/pone.0217207.zip`
1. Visit http://journals.vagrant.test/plosone/article?id=10.1371/journal.pone.0213096
1. Visit http://journals.vagrant.test/plosone/article?id=10.1371/journal.pone.0217207

image
image

## Note
The decrease in test coverage seems to be some strange artifact of using google auto-value or something? I have excluded auto-generated code from the test coverage and now the test coverage percentage has gone down. Please use your own judgment as to whether or not this code is sufficiently tested.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -626,6 +632,19 @@
...
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>0.7.5</version>
+        <configuration>
+          <excludes>
+            <!-- Simple autovalue classes -->
+            <exclude>org/ambraproject/wombat/model/Amendment.class</exclude>
```
Comment (sgatchell) [inline pom.xml]: Now I know where to stash future untested code.  :)

Comment (egh) [reply pom.xml]: Unfortunately this doesn't actually work in the jacoco run on teamcity :(

File: src/main/java/org/ambraproject/wombat/model/RelatedArticleType.java [hunk 2]
```diff
@@ -21,19 +21,23 @@
...
 import com.google.common.collect.ImmutableSet;
+import org.apache.commons.lang3.tuple.Pair;
 
 @AutoValue
 public abstract class RelatedArticleType implements Comparable<RelatedArticleType> {
+  public abstract Builder toBuilder();
   public abstract int getSortOrder();
   public abstract String getDisplayName();
   public abstract String getName();
+  // Optional would be better, but it is not well-supported in freemarker.
```
Comment (sgatchell) [inline src/main/java/org/ambraproject/wombat/model/RelatedArticleType.java]: thanks for the tip

**Comment by egh** [comment]
mobile!!!

**Comment by egh** [comment]
fixed mobile in 54976ed70

**Comment by egh** [comment]
@sgatchell Improved the mobile display
image
image

**Pull request merged**